docs/connector-development/connector-builder-ui/tutorial.md

```diff
@@ -0,0 +1,202 @@
+* On top of the form, change the "Path URL" input to `/exchangerates_data/{{ stream_slice.start_time }}` to inject the date to fetch data for into the path of the request
+* Open the "Advanced" section and set "Step" to `P1D` to configure the connector to do one separate request per day
+* Set a start date (like `2023-03-03`) in the "Testing values" menu
+* Hit the "Test" button to trigger a new test read
+
+Now, you should see a dropdown above the records view that lets you step through the daily exchange rates along with the requests performed to fetch this data. Note that in the connector builder at most 5 slices are requested to speed up testing. During a proper sync the full time range between your configured start date and the current day will be executed.
+
+When used in a connection, the connector will make sure exchange rates for the same day are not requested multiple times - the date of the last fetched record will be stored and the next scheduled sync will pick up from where the previous one stopped.
+
+## Adding transformations
+
+Note that a warning icon should show next to the "Detected schema" tab - using the per-date endpoint instead of the latest endpoint slightly changed the shape of the records by adding a `historical` property. As we don't need this property in our destination, we can remove it using a transformation.
+
+To do so, follow these steps:
+* Enable the "Transformations" section
+* Add a new transformation
+* Set the "Path" to `historical`
+* Trigger a new test read
+
+The `historical` property in the records tab and the schema warning should disappear.
+
+## Publishing and syncing
+
+So far, the connector is only configured as part of the connector builder project. To make it possible to use it in actual connections, you need to publish the connector. This captures the current state of the configuration and makes the connector available as a custom connector within the current Airbyte workspace.
+
+To use the connector for a proper sync, follow these steps:
+* Click the "Publish" button and publish the first version of the "Stock Exchange" connector
+* Go to the "Connections" page and create a new connection
+* As Source type, pick the "Stock Exchange" connector you just created
+* Set API Key, base currency and start date for the sync - to avoid a large number of requests, set the start date to one week in the past
+* Click "Set up source" and wait for the connection check to validate the provided configuration is valid
+* Set up a destination - to keep things simple let's choose the "E2E Testing" destination type
+* Click "Set up destination", keeping the default configurations
+* Wait for Airbyte to check the record schema, then click "Set up connection" - this will create the connection and kick off the first sync
+* After a short while, the sync should complete successfully
```

Comment:
When I did this, I opened the logs of my job to see the output record and I instead just found a bunch of these deserialization errors:
<img width="2197" alt="Screen Shot 2023-04-14 at 5 39 37 PM" src="https://user-images.githubusercontent.com/22731524/232174593-bc10d061-3fd1-4b69-a1a4-3259fc2b111b.png">

A cursory search on slack revealed this may be because the records are large: https://airbytehq-team.slack.com/archives/C048E3BK1KM/p1680822782177669?thread_ts=1680814742.350359&cid=C048E3BK1KM

It would be nice if users didn't see a bunch of error messages during this tutorial - maybe we can ask the platform team for help here? Only other alternatives I'm imagining are using a different connector with smaller output records (but that would mean rewriting most of this tutorial) or trying to configure the record selector to somehow split up the rates into separate records, but I couldn't figure out an easy way to do that

Reply:
I don't think this is related to the size of the record, our records are fairly small.

I think it's a bug of the e2e destination - the platform tries to interpret its log messages as airbyte messages:
```
2023-04-17 11:10:13 ERROR i.a.w.i.VersionedAirbyteStreamFactory(toAirbyteMessage):360 - Deserialization failed: "2023-04-17 11:10:13 \u001B[32mINFO\u001B[m i.a.i.d.e.TestingDestinations(main):71 - starting destination: class io.airbyte.integrations.destination.e2e_test.TestingDestinations"
```

You can see that the log message `starting destination: class io.airbyte.integrations.destination.e2e_test.TestingDestinations` gets parsed as an Airbyte message which fails. I'm not sure who can help, I will ask the destinations team.

I think we should fix this bug in the e2e connector rather than switching away from it as there is no destination that works without configuration which keeps the tutorial focused.